### ISSUE TITLE
event should support IAM authorized API Gateway event

### ISSUE THREAD

**Issue by u873838** [astuyve/lambda-stream#16]
The event argument currently has a fixed type of APIGatewayProxyEventV2. This should be expanded to allow to support at least APIGatewayProxyEventV2WithIAMAuthorizer, which is the event type when invoking via a function URL with IAM authorization.

**Comment by astuyve** [comment]
Yup totally, please submit a PR and I'll merge and release it. Just @ mention me so I see it, thanks!
